creusot/src/translation/pearlite.rs

```diff
@@ -225,8 +225,9 @@ pub(crate) fn pearlite<'tcx>(
         return Err(Error::new(ctx.def_span(id), "type checking failed"));
     };
 
+    // let purity = Purity::of_def_id(ctx.tcx, id.to_def_id());
     visit::walk_expr(
```

Comment:
Why do we have to check purity again here? Shouldn't `before_analysis` already check every HIR item?

Reply:
hmm, i think its a relic of times long past.